### PR TITLE
Utility::parseCookieValue

### PR DESCRIPTION
**Pull request by wgallagher**

### CODE REVIEW

File: source/common/http/utility.cc [hunk 1]
```diff
@@ -61,6 +61,34 @@ Utility::QueryParams Utility::parseQueryString(const std::string& url) {
   return params;
 }
 
+std::string Utility::parseCookieValue(const HeaderMap& headers, const std::string& key) {
+
+  struct State {
+    std::string key_;
+    std::string ret_{};
```
Comment (mattklein123) [inline source/common/http/utility.cc]: nit: don't need {}

**Comment by mattklein123** [comment]
@wgallagher this looks fine to me. Can you have a web person take a quick look at this. Perhaps @heston ?

**Comment by mattklein123** [comment]
@saponifi3d @dschaller can one of you take a look at this please. We just want to make sure the test cases cover what needs to be covered and things look sane.

**Comment by dschaller** [comment]
Looks good to me. Talked with @wgallagher offline and as long as the cookies are being passed in as headers this should be good.

**Comment by RyanMcG** [comment]
@wgallagher looks good to me.

**Pull request merged**